### ISSUE TITLE
Error during compiling openMVG_main_evalQuality on MacOs 10.13.1

### ISSUE THREAD

**Issue by nicolapiccinelli** [openMVG/openMVG#1155]
Hi there, i'm using openMVG on tag v1.2 but after the update to MacOs 10.13.1 i'm not able to complete the build anymore:

```
CompileC /Users/nicolapiccinelli/Documents/MATLAB/FSE/deps/openMVG_Build/software/SfM/openMVG.build/Release/openMVG_main_evalQuality.build/Objects-normal/x86_64/main_evalQuality.o software/SfM/main_evalQuality.cpp normal x86_64 c++ com.apple.compilers.llvm.clang.1_0.compiler
    cd /Users/nicolapiccinelli/Documents/MATLAB/FSE/deps/openMVG/src
    export LANG=en_US.US-ASCII
    /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang -x c++ -arch x86_64 -fmessage-length=208 -fdiagnostics-show-note-include-stack -fmacro-backtrace-limit=0 -fcolor-diagnostics -Wno-trigraphs -fpascal-strings -O3 -Wno-missing-field-initializers -Wno-missing-prototypes -Wno-return-type -Wno-non-virtual-dtor -Wno-overloaded-virtual -Wno-exit-time-destructors -Wno-missing-braces -Wparentheses -Wswitch -Wno-unused-function -Wno-unused-label -Wno-unused-parameter -Wno-unused-variable -Wunused-value -Wno-empty-body -Wno-uninitialized -Wno-unknown-pragmas -Wno-shadow -Wno-four-char-constants -Wno-conversion -Wno-constant-conversion -Wno-int-conversion -Wno-bool-conversion -Wno-enum-conversion -Wno-float-conversion -Wno-non-literal-null-conversion -Wno-objc-literal-conversion -Wno-shorten-64-to-32 -Wno-newline-eof -Wno-c++11-extensions -DCMAKE_INTDIR=\"Release\" -DOPENMVG_STD_UNORDERED_MAP -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.13.sdk -fasm-blocks -fstrict-aliasing -Wdeprecated-declarations -Winvalid-offsetof -mmacosx-version-min=10.13 -Wno-sign-conversion -Wno-infinite-recursion -Wno-move -Wno-comma -Wno-block-capture-autoreleasing -Wno-strict-prototypes -Wno-range-loop-analysis -I/Users/nicolapiccinelli/Documents/MATLAB/FSE/deps/openMVG_Build/Darwin-x86_64-RELEASE/Release/include -I/Users/nicolapiccinelli/Documents/MATLAB/FSE/deps/openMVG/src/third_party/eigen -I/Users/nicolapiccinelli/Documents/MATLAB/FSE/deps/openMVG/src -I/Users/nicolapiccinelli/Documents/MATLAB/FSE/deps/openMVG/src/dependencies -I/Users/nicolapiccinelli/Documents/MATLAB/FSE/deps/openMVG/src/dependencies/cereal/include -I/Users/nicolapiccinelli/Documents/MATLAB/FSE/deps/openMVG/src/third_party/lemon -I/Users/nicolapiccinelli/Documents/MATLAB/FSE/deps/openMVG_Build/third_party/lemon -I/Users/nicolapiccinelli/Documents/MATLAB/FSE/deps/openMVG/src/third_party/flann/src/cpp -I/Users/nicolapiccinelli/Documents/MATLAB/FSE/deps/openMVG_Build/software/SfM/openMVG.build/Release/openMVG_main_evalQuality.build/DerivedSources/x86_64 -I/Users/nicolapiccinelli/Documents/MATLAB/FSE/deps/openMVG_Build/software/SfM/openMVG.build/Release/openMVG_main_evalQuality.build/DerivedSources -Wmost -Wno-four-char-constants -Wno-unknown-pragmas -F/Users/nicolapiccinelli/Documents/MATLAB/FSE/deps/openMVG_Build/Darwin-x86_64-RELEASE/Release -march=sandybridge -msse2 -msse3 -mssse3 -msse4.1 -msse4.2 -mavx -mno-sse4a -mno-fma -mno-bmi2 -mno-avx2 -mno-xop -mno-fma4 -mno-avx512f -mno-avx512vl -mno-avx512pf -mno-avx512er -mno-avx512cd -mno-avx512dq -mno-avx512bw -mno-avx512ifma -mno-avx512vbmi -std=c++11 -MMD -MT dependencies -MF /Users/nicolapiccinelli/Documents/MATLAB/FSE/deps/openMVG_Build/software/SfM/openMVG.build/Release/openMVG_main_evalQuality.build/Objects-normal/x86_64/main_evalQuality.d --serialize-diagnostics /Users/nicolapiccinelli/Documents/MATLAB/FSE/deps/openMVG_Build/software/SfM/openMVG.build/Release/openMVG_main_evalQuality.build/Objects-normal/x86_64/main_evalQuality.dia -c /Users/nicolapiccinelli/Documents/MATLAB/FSE/deps/openMVG/src/software/SfM/main_evalQuality.cpp -o /Users/nicolapiccinelli/Documents/MATLAB/FSE/deps/openMVG_Build/software/SfM/openMVG.build/Release/openMVG_main_evalQuality.build/Objects-normal/x86_64/main_evalQuality.o
In file included from /Users/nicolapiccinelli/Documents/MATLAB/FSE/deps/openMVG/src/software/SfM/main_evalQuality.cpp:9:
In file included from /Users/nicolapiccinelli/Documents/MATLAB/FSE/deps/openMVG/src/openMVG/sfm/sfm_data.hpp:14:
In file included from /Users/nicolapiccinelli/Documents/MATLAB/FSE/deps/openMVG/src/openMVG/cameras/Camera_Intrinsics.hpp:15:
In file included from /Users/nicolapiccinelli/Documents/MATLAB/FSE/deps/openMVG/src/openMVG/geometry/pose3.hpp:12:
In file included from /Users/nicolapiccinelli/Documents/MATLAB/FSE/deps/openMVG/src/openMVG/multiview/projection.hpp:32:
In file included from /Users/nicolapiccinelli/Documents/MATLAB/FSE/deps/openMVG/src/openMVG/numeric/eigen_alias_definition.hpp:18:
In file included from /Users/nicolapiccinelli/Documents/MATLAB/FSE/deps/openMVG/src/third_party/eigen/Eigen/SparseCore:16:
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/map:820:5: error: static_assert failed "Allocator::value_type must be same type as value_type"
    static_assert((is_same<typename allocator_type::value_type, value_type>::value),
    ^             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /Users/nicolapiccinelli/Documents/MATLAB/FSE/deps/openMVG/src/software/SfM/main_evalQuality.cpp:12:
/Users/nicolapiccinelli/Documents/MATLAB/FSE/deps/openMVG/src/software/SfM/io_readGT.hpp:158:24: note: in instantiation of template class 'std::__1::map<unsigned long, openMVG::cameras::PinholeCamera,
      std::__1::less<unsigned long>, Eigen::aligned_allocator<std::__1::pair<unsigned long, openMVG::cameras::PinholeCamera> > >' requested here
          map_camerasGT.insert({std::distance(vec_camfilenames.cbegin(), iter), cam});
                       ^
/Users/nicolapiccinelli/Documents/MATLAB/FSE/deps/openMVG/src/software/SfM/main_evalQuality.cpp:122:29: error: no member named 'size' in 'std::__1::map<unsigned long, openMVG::cameras::PinholeCamera,
      std::__1::less<unsigned long>, Eigen::aligned_allocator<std::__1::pair<unsigned long, openMVG::cameras::PinholeCamera> > >'
    std::cout << map_Cam_gt.size() << " gt cameras have been found" << std::endl;
                 ~~~~~~~~~~ ^
/Users/nicolapiccinelli/Documents/MATLAB/FSE/deps/openMVG/src/software/SfM/main_evalQuality.cpp:133:18: error: no member named 'size' in 'std::__1::map<unsigned long, openMVG::cameras::PinholeCamera,
      std::__1::less<unsigned long>, Eigen::aligned_allocator<std::__1::pair<unsigned long, openMVG::cameras::PinholeCamera> > >'
  if (map_Cam_gt.size() != sfm_data.GetPoses().size())
      ~~~~~~~~~~ ^
/Users/nicolapiccinelli/Documents/MATLAB/FSE/deps/openMVG/src/software/SfM/main_evalQuality.cpp:137:38: error: no member named 'size' in 'std::__1::map<unsigned long, openMVG::cameras::PinholeCamera,
      std::__1::less<unsigned long>, Eigen::aligned_allocator<std::__1::pair<unsigned long, openMVG::cameras::PinholeCamera> > >'
      << "#GT poses: " << map_Cam_gt.size() << "\n"
                          ~~~~~~~~~~ ^
/Users/nicolapiccinelli/Documents/MATLAB/FSE/deps/openMVG/src/software/SfM/main_evalQuality.cpp:146:77: error: no member named 'begin' in 'std::__1::map<unsigned long, openMVG::cameras::PinholeCamera,
      std::__1::less<unsigned long>, Eigen::aligned_allocator<std::__1::pair<unsigned long, openMVG::cameras::PinholeCamera> > >'
  for (std::map< size_t, PinholeCamera>::const_iterator iterGT = map_Cam_gt.begin();
                                                                 ~~~~~~~~~~ ^
/Users/nicolapiccinelli/Documents/MATLAB/FSE/deps/openMVG/src/software/SfM/main_evalQuality.cpp:147:26: error: no member named 'end' in 'std::__1::map<unsigned long, openMVG::cameras::PinholeCamera,
      std::__1::less<unsigned long>, Eigen::aligned_allocator<std::__1::pair<unsigned long, openMVG::cameras::PinholeCamera> > >'
    iterGT != map_Cam_gt.end(); ++iterGT, ++iter_loaded_poses)
              ~~~~~~~~~~ ^
6 errors generated.

** BUILD FAILED **


The following build commands failed:
	CompileC /Users/nicolapiccinelli/Documents/MATLAB/FSE/deps/openMVG_Build/software/SfM/openMVG.build/Release/openMVG_main_evalQuality.build/Objects-normal/x86_64/main_evalQuality.o software/SfM/main_evalQuality.cpp normal x86_64 c++ com.apple.compilers.llvm.clang.1_0.compiler
(1 failure)
```

This is the compiler i've actually on my computer:

```
Apple LLVM version 9.0.0 (clang-900.0.38)
Target: x86_64-apple-darwin17.2.0
Thread model: posix
InstalledDir: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin
```

**Comment by nicolapiccinelli** [comment]
Update, Ii've tried to remove the **openMVG_main_evalQuality** from the CMakeLists for the SfM softwares and the build succeeded.

**Comment by pmoulon** [comment]
Thx for the reporting.
This error is fixed in the develop branch (coming version 1.3).
